dev/changelog/22.0.0.md

```diff
@@ -0,0 +1,141 @@
+<!---
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing,
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+-->
+
+## [22.0.0](https://github.com/apache/arrow-datafusion/tree/22.0.0) (2023-04-07)
+
+[Full Changelog](https://github.com/apache/arrow-datafusion/compare/21.1.0...22.0.0)
+
+**Breaking changes:**
+
+- Introduce a common trait TreeNode for ExecutionPlan, PhysicalExpr, LogicalExpr, LogicalPlan [#5630](https://github.com/apache/arrow-datafusion/pull/5630) (yahoNanJing)
+- Minor: Reduce clones in AnalyzerRule [#5728](https://github.com/apache/arrow-datafusion/pull/5728) (alamb)
+- Change required input ordering physical plan API to allow any NULLS FIRST / LAST and ASC / DESC [#5772](https://github.com/apache/arrow-datafusion/pull/5772) (mustafasrepo)
```

Comment:
I would like to try and get https://github.com/apache/arrow-datafusion/pull/5863 in (to minimize the external API churn). If it doesn't make ti that is ok

Reply:
I am no rush, and generating the changelog is super fast these days